opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3/v3_2_0/TASK_5964/AddNewClinicalStatusValues.java

```diff
@@ -0,0 +1,110 @@
+                        bulk.add(new UpdateOneModel<>(Filters.eq("_id", document.get("_id")), effectiveUpdate));
+                    });
+        }
+
+        for (String collection : Arrays.asList(OrganizationMongoDBAdaptorFactory.INTERPRETATION_COLLECTION,
+                OrganizationMongoDBAdaptorFactory.INTERPRETATION_ARCHIVE_COLLECTION,
+                OrganizationMongoDBAdaptorFactory.DELETED_INTERPRETATION_COLLECTION)) {
+            migrateCollection(collection, query, projection,
+                    (document, bulk) -> {
+                        MongoDBAdaptor.UpdateDocument updateDocument = new MongoDBAdaptor.UpdateDocument();
+                        String interpretationId = document.getString("id");
+                        Document status = document.get("status", Document.class);
+
+                        Document newStatus = fillStatusValues("Interpretation", interpretationId, status,
+                                defaultConfiguration.getInterpretation().getStatus());
+                        updateDocument.getSet().put("status", newStatus);
+
+                        Document effectiveUpdate = updateDocument.toFinalUpdateDocument();
+
+                        logger.debug("Updating interpretation '{}': {}", document.get("id"), effectiveUpdate.toBsonDocument());
+                        bulk.add(new UpdateOneModel<>(Filters.eq("_id", document.get("_id")), effectiveUpdate));
+                    });
+        }
+    }
+
+    private Document fillStatusValues(String entity, String id, Document status, List<ClinicalStatusValue> statusValueList) {
+        ClinicalStatus clinicalStatus = new ClinicalStatus();
+        String clinicalId = status != null ? status.getString("id") : null;
+        if (status == null || StringUtils.isEmpty(clinicalId)) {
+            logger.warn("Status is empty or does not contain 'id' field. Setting default status value for {} '{}'", entity, id);
+            for (ClinicalStatusValue clinicalStatusValue : statusValueList) {
+                if (clinicalStatusValue.getType().equals(ClinicalStatusValue.ClinicalStatusType.NOT_STARTED)) {
+                    clinicalStatus.setId(clinicalStatusValue.getId());
+                    clinicalStatus.setDescription(clinicalStatusValue.getDescription());
+                    clinicalStatus.setType(clinicalStatusValue.getType());
+                }
+            }
+        } else {
+            for (ClinicalStatusValue clinicalStatusValue : statusValueList) {
+                if (clinicalStatusValue.getId().equals(clinicalId)) {
+                    clinicalStatus.setId(clinicalStatusValue.getId());
+                    clinicalStatus.setDescription(clinicalStatusValue.getDescription());
+                    clinicalStatus.setType(clinicalStatusValue.getType());
+                    break;
+                }
+            }
+            if (clinicalStatus.getType() == null) {
```

Comment:
Risky point here... we'd overwrite/remove the old values

Reply:
That scenario is extremely unlikely but I will change to keep the original values instead of removing them. You're right